numpy/core/src/multiarray/dtype_traversal.c

```diff
@@ -271,10 +314,11 @@ traverse_fields_function(
 
 
 static int
-get_clear_fields_transfer_function(
+get_fields_traverse_function(
         void *traverse_context, PyArray_Descr *dtype, int NPY_UNUSED(aligned),
         npy_intp stride, traverse_loop_function **out_func,
-        NpyAuxData **out_auxdata, NPY_ARRAYMETHOD_FLAGS *flags)
+        NpyAuxData **out_auxdata, NPY_ARRAYMETHOD_FLAGS *flags,
+        traverse_func_get *traverse_func_get)
```

Comment:
Looks a bit odd the use the same name for the type and the variable.

Reply:
Also below.

Reply:
Good point, a bit clunky but renamed the type to `get_traverse_func_function` and the function use to `get_traverse_func` (get first is mostly used).